xdsl/interactive/app.py

```diff
@@ -135,16 +135,27 @@ class InputApp(App[None]):
     passes_list_view: ListView
     """ListView displaying the passes available to apply."""
 
-    input_number_of_ops: DataTable[str | int]
-    output_number_of_ops: DataTable[str | int]
+    input_operation_count_dict: dict[str, int] = dict()
+    """Saves the operation name and count of the input text area in a dictionary."""
+    output_operation_count_dict: dict[str, int] = dict()
+    """Saves the operation name and count of the output text area in a dictionary."""
```

Comment:
ah, this is exactly what I was hoping for in my comment in the other PR, could you please move this change there? One bug here is that dict is mutable, so changes that mutate it directly won't be picked up by the reactive system, could you please make it an n-tuple of pairs, to make sure that if it's mutated the system propagates the changes?

Reply:
Yes will do.